miniapps/parelag/MultilevelHcurlSolver.cpp

```diff
@@ -0,0 +1,739 @@
+// Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include <fstream>
+#include <sstream>
+#include <ostream>
+#include <string>
+#include <vector>
+#include <memory>
+
+#include <mpi.h>
+
+#include "elag.hpp"
+#include "utilities/MPIDataTypes.hpp"
+
+using namespace mfem;
+using namespace parelag;
+using namespace std;
+
+void bdrfunc(const Vector &, Vector &);
+void rhsfunc(const Vector &, Vector &);
+
+int main(int argc, char *argv[])
+{
+   // Initialize MPI.
+   mpi_session session(argc, argv);
+   MPI_Comm comm = MPI_COMM_WORLD;
+   int num_ranks, myid;
+   MPI_Comm_size(comm, &num_ranks);
+   MPI_Comm_rank(comm, &myid);
+
+   Timer total_timer = TimeManager::AddTimer("Program Execution -- Total");
+   Timer init_timer = TimeManager::AddTimer("Initial Setup");
+
+   if (!myid)
+      cout << "-- This is an example of using a geometric-like multilevel "
+           "hierarchy, constructed by ParElag,\n"
+           "-- to solve a finite element H(curl) form: "
+           "(alpha curl u, curl v) + (beta u, v).\n\n";
+
+   // Get basic parameters from command line.
+   const char *xml_file_c = "MultilevelHcurlSolver_cube_example_parameters.xml";
+   bool visualize = false;
+   double tolSVD = 1e-3;
```

Comment:
It would be helpful to briefly document what this SVD tolerance is for.

Reply:
I'll add more information in the respective `args.AddOption()`